sql_runner/snowflake_target.go

```diff
@@ -49,31 +66,95 @@ func (sft SnowFlakeTarget) GetTarget() Target {
 
 // Run a query against the target
 // One statement per API call
-func (sft SnowFlakeTarget) RunQuery(query ReadyQuery, dryRun bool) QueryStatus {
+func (sft SnowFlakeTarget) RunQuery(query ReadyQuery, dryRun bool, showQueryOutput bool) QueryStatus {
 	var affected int64 = 0
 	var err error
 
 	if dryRun {
+		if sft.IsConnectable() {
+			log.Printf("SUCCESS: Able to connect to target database, %s\n.", sft.Account)
+		} else {
+			log.Printf("ERROR: Cannot connect to target database, %s\n.", sft.Account)
+		}
+
 		return QueryStatus{query, query.Path, 0, nil}
 	}
 
 	scripts := strings.Split(query.Script, ";")
 
 	for _, script := range scripts {
-		var res sql.Result
-
 		if len(strings.TrimSpace(script)) > 0 {
-
-			res, err = sft.Client.Exec(script)
-
-			if err != nil {
-				return QueryStatus{query, query.Path, int(affected), err}
+			if showQueryOutput {
+				rows, err := sft.Client.Query(script)
+				if err != nil {
+					log.Printf("ERROR: %s.", err)
+					return QueryStatus{query, query.Path, int(affected), err}
+				}
+
+				err = printSfTable(rows)
+				if err != nil {
+					log.Printf("ERROR: %s.", err)
+					return QueryStatus{query, query.Path, int(affected), err}
+				}
 			} else {
-				aff, _ := res.RowsAffected()
-				affected += aff
+				res, err := sft.Client.Exec(script)
+
+				if err != nil {
+					return QueryStatus{query, query.Path, int(affected), err}
+				} else {
+					aff, _ := res.RowsAffected()
+					affected += aff
+				}
```

Comment:
this seems very similar to pg, can't this be abstracted

Reply:
Good point. I'll try to refactor before release.